quaint/src/single.rs

```diff
@@ -16,7 +16,7 @@ use crate::connector::NativeConnectionInfo;
 /// The main entry point and an abstraction over a database connection.
 #[derive(Clone)]
 pub struct Quaint {
-    inner: Arc<dyn Queryable>,
+    inner: Arc<dyn TransactionCapable>,
```

Comment:
Am I right in thinking that because the inner was `Queryable` and not `TransactionCapable` the implementation was not being used correctly as the `Mssql::start_transaction` method is for `TransactionCapable` not `Queryable`?

Reply:
It was because `Quaint::start_transaction` and `PooledConnection::start_transaction` never actually called the `start_transaction` method of the underlying queryable at all, they had their own, default, implementations on top. This PR changes it to forward the calls instead, and it requires changing `dyn Queryable` to `dyn TransactionCapable` first so that the method is available to call.